src/sp/protocol/mqtt/mqtt_parser.c

```diff
@@ -586,6 +586,9 @@ conn_handler(uint8_t *packet, conn_param *cparam)
 		property *assigned_cid =
 		    property_set_value_str(ASSIGNED_CLIENT_IDENTIFIER,
 		        cparam->clientid.body, cparam->clientid.len, false);
+		if (cparam->properties == NULL) {
+			cparam->properties = property_alloc();
```

Comment:
I feel confused why alloc can fix memory leak.

Reply:
Becuase _property_set_value_str()_ will alloc new memory for  _assigned_cid_ . If _cparam->properties_ is _NULL_, then property_free() is unable  to free _assigned_cid_.

Reply:
If cparam->properties is NULL, should property_append return error ?